### ISSUE TITLE
Possible typo in `copy!(x::AbstractVector, ps::Params)`

### ISSUE THREAD

**Issue by pakk-minidose** [FluxML/Zygote.jl#1135]
Hello,
the following function
https://github.com/FluxML/Zygote.jl/blob/601f888c701d21971c9857adad90318cd8c3ea10/src/compiler/interface.jl#L231

should return `x` instead of `ps`, in order to be consistent with the Julia documentation, which states that the first argument should be returned

```
help?> copy!
search: copy! copyto! circcopy! unsafe_copyto! copy copysign deepcopy

  copy!(dst, src) -> dst

  In-place copy of src into dst, discarding any pre-existing elements in dst. If dst and src are of the same type, dst == src should hold after the call. If dst and src are 
  multidimensional arrays, they must have equal axes. See also copyto!.
```

Is this a typo, or is it intentional? I wanted to ask before submitting a PR.

Thank you for your reply.

**Comment by DhairyaLGandhi** [comment]
There isn't a strict requirement and returns on mutating functions are mostly for convenience but in this case it does seem better to return `x` instead.

**Comment by ToucheSir** [comment]
Judging by the original commit, it may well have been a copy paste typo. Please do file a PR!

**Comment by pakk-minidose** [comment]
Thank you for your replies!

@ToucheSir done!
@DhairyaLGandhi I have expanded on your point in the PR

**Comment by mcabbott** [comment]
Note, for future, that "closes #1135" has to be in the first post for github to notice it. But this is fixed, thanks.
